src/rascal/utils.hh

```diff
@@ -38,6 +38,62 @@
 namespace rascal {
   namespace internal {
 
+    namespace details {
```

Comment:
Not very important, but why do we need another level of namespace here? We are already in `internal`

Reply:
because it is a detail that is only used in this internal bracket